src/dimension.c

```diff
@@ -441,3 +519,431 @@ hyperspace_calculate_point(Hyperspace *hs, HeapTuple tuple, TupleDesc tupdesc)
+		(coltype == INT4OID && value > INT32_MAX) ||
+		value < 1)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("invalid interval: must be between 1 and " INT64_FORMAT,
+						INT_TYPE_MAX(coltype))));
+
+	if (show_warnings && IS_TIMESTAMP_TYPE(coltype) && value < USECS_PER_SEC)
+		ereport(WARNING,
+				(errcode(ERRCODE_AMBIGUOUS_PARAMETER),
+				 errmsg("unexpected interval: smaller than one second"),
+				 errhint("The interval is specified in microseconds")));
+
+	return value;
+}
+
+static int64
+dimension_interval_to_internal(Oid coltype, Oid valuetype, Datum value, bool show_warnings)
+{
+	switch (valuetype)
+	{
+		case INT2OID:
+			return get_validated_integer_interval(coltype, DatumGetInt16(value), show_warnings);
+		case INT4OID:
+			return get_validated_integer_interval(coltype, DatumGetInt32(value), show_warnings);
+		case INT8OID:
+			return get_validated_integer_interval(coltype, DatumGetInt64(value), show_warnings);
+		case INTERVALOID:
+			if (IS_INTEGER_TYPE(coltype))
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("invalid interval: must be an integer type for integer dimensions")));
+
+			return interval_to_usec(DatumGetIntervalP(value));
+		default:
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 errmsg("invalid interval: must be an interval or integer type")));
+	}
+}
+
+static void
+dimension_add_not_null_on_column(Oid table_relid, char *colname)
+{
+	AlterTableCmd cmd = {
+		.type = T_AlterTableCmd,
+		.subtype = AT_SetNotNull,
+		.name = colname,
+		.missing_ok = false,
+	};
+
+	elog(NOTICE, "Adding NOT NULL constraint to time column %s (NULL time values not allowed)", colname);
+
+	AlterTableInternal(table_relid, list_make1(&cmd), false);
+}
+
+static void
+dimension_update(FunctionCallInfo fcinfo,
+				 Oid table_relid,
+				 Name dimname,
+				 DimensionType dimtype,
+				 Datum *interval,
+				 int16 *num_slices)
+{
+	Cache	   *hcache = hypertable_cache_pin();
+	Hypertable *ht;
+	Dimension  *dim;
+
+	if (dimtype == DIMENSION_TYPE_ANY)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("invalid dimension type")));
+
+	ht = hypertable_cache_get_entry(hcache, table_relid);
+
+	if (NULL == ht)
+		ereport(ERROR,
+				(errcode(ERRCODE_IO_HYPERTABLE_NOT_EXIST),
+				 errmsg("table \"%s\" is not a hypertable",
+						get_rel_name(table_relid))));
+
+	if (NULL == dimname)
+	{
+		if (hyperspace_get_num_dimensions_by_type(ht->space, dimtype) > 1)
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 errmsg("hypertable \"%s\" has multiple %s dimensions",
+							get_rel_name(table_relid),
+							dimtype == DIMENSION_TYPE_OPEN ? "time" : "space"),
+					 errhint("An explicit dimension name needs to be specified")));
+
+		dim = hyperspace_get_dimension(ht->space, dimtype, 0);
+	}
+	else
+		dim = hyperspace_get_dimension_by_name(ht->space, dimtype, NameStr(*dimname));
```

Comment:
Not sure what you mean. Do you want an `Assert()`? The type is given as argument when retrieving the dimension. Do you want to test that `hyperspace_get_dimension()` doesn't have a bug? In that case I think we should unit test that. But I am not sure why this function should be singled out since it is quite simple?

Reply:
ahh didn't see the dimtype arg